src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -399,14 +399,15 @@ public static string ConcatAsHexSuffix(string str, char separator, uint number)
                     i = tupleStr.Length;
                 }
 
-                buffer[i + 8] = encode16Chars[tupleNumber & 0xF];
-                buffer[i + 7] = encode16Chars[(tupleNumber >> 4) & 0xF];
-                buffer[i + 6] = encode16Chars[(tupleNumber >> 8) & 0xF];
-                buffer[i + 5] = encode16Chars[(tupleNumber >> 12) & 0xF];
-                buffer[i + 4] = encode16Chars[(tupleNumber >> 16) & 0xF];
-                buffer[i + 3] = encode16Chars[(tupleNumber >> 20) & 0xF];
-                buffer[i + 2] = encode16Chars[(tupleNumber >> 24) & 0xF];
-                buffer[i + 1] = encode16Chars[(tupleNumber >> 28) & 0xF];
+                var number = (int)tupleNumber;
+                buffer[i + 8] = (char)hexEncodeMap[number & 0xF];
+                buffer[i + 7] = (char)hexEncodeMap[(number >> 4) & 0xF];
+                buffer[i + 6] = (char)hexEncodeMap[(number >> 8) & 0xF];
+                buffer[i + 5] = (char)hexEncodeMap[(number >> 12) & 0xF];
+                buffer[i + 4] = (char)hexEncodeMap[(number >> 16) & 0xF];
+                buffer[i + 3] = (char)hexEncodeMap[(number >> 20) & 0xF];
+                buffer[i + 2] = (char)hexEncodeMap[(number >> 24) & 0xF];
+                buffer[i + 1] = (char)hexEncodeMap[(number >> 28) & 0xF];
                 buffer[i] = tupleSeparator;
```

Comment:
it can be vectorized I think, load `tupleNumber` into an 256bit vector, do some magic and dump to a string (ideally - to FastAllocateString):
![image](https://user-images.githubusercontent.com/523221/72614135-89c16880-3942-11ea-8f76-ef8399787609.png)

Reply:
`FastAllocateString` is `internal` so it can't be used here in ASP.NET Core?

256bit vec is too large, 128bit should work, as there get 9 elements written to `buffer`. So with `Vector128<ushor>` it's 8 elements + the remaining 1.

Preferable this should be done in a different PR -- but nice idea!

Reply:
That's just widening it, not converting to ascii hex?

Reply:
@gfoidl yeah I meant Vector128 :-)

Reply:
@gfoidl @benaadams here is a vectorized `string ToHex(int num)` : https://pastebin.com/F2ZUYkwG

Reply:
Looks good.

Makes it sense to add `int -> hex` to runtime? I think it's quite common to convert a number to hex. Therefore is `int.ToString("X")`, but it allocates a string. So there should be something like `int.ToString(Span<char>, Format)` (I know the API needs work, maybe on a better type like a special formatter) and at this central place all kind of optimization could take place (SIMD or SWAR).

Reply:
> Makes it sense to add int -> hex to runtime? 

That conversation has been going on for years https://github.com/dotnet/corefx/issues/10013

Reply:
Gowan then
```
            Method | Branch |     Mean |     Error |         Op/s | Allocated | Precent |
------------------ |--------|---------:|----------:|-------------:|----------:|--------:|
 ConcatAsHexSuffix | master | 21.18 ns | 0.1040 ns | 47,221,986.9 |  34.33 MB |         |
 ConcatAsHexSuffix |     pr | 17.26 ns | 0.0844 ns | 57,953,720.1 |  34.33 MB |   123 % |
```

Reply:
I have to search within my brain, but I believe there is a SWAR variant that needs less instructions than this SIMD variant. I'll give an update here...

Reply:
> That conversation has been going on for years

Maybe split the hex-part out of `PopulateSpanWithHexSuffix` to have such a method at least here in aspnetcore?